internal/armada/configuration/types.go

```diff
@@ -97,13 +97,21 @@ type SchedulingConfig struct {
 	Lease                                     LeaseSettings
 	DefaultJobLimits                          common.ComputeResources
 	DefaultJobTolerations                     []v1.Toleration
+	TerminationGracePeriod                    TerminationGracePeriodConfig
 	MaxRetries                                uint // Maximum number of retries before a Job is failed
 	ResourceScarcity                          map[string]float64
 	PoolResourceScarcity                      map[string]map[string]float64
 	MaxPodSpecSizeBytes                       uint
 	MinJobResources                           v1.ResourceList
 }
 
+// TerminationGracePeriodConfig applied to preemptible jobs
```

Comment:
So the comment says that this is for preempible jobs only, but the configuration is at the Scheduling level (i.e. `SchedulingConfig.TerminationGracePeriod  ` rather than at the Preemption level (i.e. `Scheduling.Preemption .TerminationGracePeriod ` )

I think we should either move this to be under `Premption` or have `TerminationGracePeriodConfig` applied to all jobs regardless.

Reply:
FWIW My personal preference would be to just change the PremptionConfig to be:
```
type PreemptionConfig struct {
	Enabled bool
	PriorityClasses map[string]int32
	DefaultPriorityClass string
	// New Fields added for Grace period
        MinTerminationGracePeriod time.Duration
	MaxTerminationGracePeriod time.Duration
	DefaulTerminationGracePeriod time.Duration
}
```

Reply:
Makes sense, can move it there